estimators/mock_data/make.py

```diff
@@ -0,0 +1,106 @@
+from gammapy.modeling.models import (SpatialModel, 
+                                     PowerLawSpectralModel, 
+                                     SkyModel)
+from gammapy.modeling import Parameter
+from gammapy.irf import load_irf_dict_from_file
+from gammapy.datasets import MapDataset
+from gammapy.data import FixedPointingInfo, Observation, observatory_locations
+from gammapy.makers import MapDatasetMaker, SafeMaskMaker
+from gammapy.maps import MapAxis, WcsGeom
+
+from astropy import units as u
+from astropy.coordinates import SkyCoord, angular_separation
+import numpy as np
+
+# To create an energy-dependent morphology we start by defining parameters for different energies
+
+class MyCustomGaussianModel(SpatialModel):
+    """My custom Energy Dependent Gaussian model.
+
+    Parameters
+    ----------
+    lon_0, lat_0 : `~astropy.coordinates.Angle`
+        Center position
+    sigma_1TeV : `~astropy.coordinates.Angle`
+        Width of the Gaussian at 1 TeV
+    sigma_10TeV : `~astropy.coordinates.Angle`
+        Width of the Gaussian at 10 TeV
+
+    """
+    tag = "MyCustomGaussianModel"
+    is_energy_dependent = True
+    lon_0 = Parameter("lon_0", "5.6 deg")
+    lat_0 = Parameter("lat_0", "0.2 deg", min=-90, max=90)
+    
+    sigma_1TeV = Parameter("sigma_1TeV", "0.3 deg", min=0)
+    sigma_10TeV = Parameter("sigma_10TeV", "0.15 deg", min=0)
+    
+    SpatialModel.frame = 'galactic'
+
+
+    @staticmethod
+    def evaluate(lon, lat, energy, lon_0, lat_0, sigma_1TeV, sigma_10TeV):
+
+        sep = angular_separation(lon, lat, lon_0, lat_0)
+
+        # Compute sigma for the given energy using linear interpolation in log energy
+        sigma_nodes = u.Quantity([sigma_1TeV, sigma_10TeV])
+        energy_nodes = [1, 10] * u.TeV
+        log_s = np.log(sigma_nodes.to("deg").value)
+        log_en = np.log(energy_nodes.to("TeV").value)
+        log_e = np.log(energy.to("TeV").value)
+        sigma = np.exp(np.interp(log_e, log_en, log_s)) * u.deg
+
+        exponent = -0.5 * (sep / sigma) ** 2
+        norm = 1 / (2 * np.pi * sigma**2)
+
+        return norm * np.exp(exponent)
+    
+    @property
+    def evaluation_radius(self):
+        """Evaluation radius (`~astropy.coordinates.Angle`)."""
+        return 2 * np.max([self.sigma_1TeV.value, self.sigma_10TeV.value]) * u.deg
+        
+# Create spatial model        
+spatial_model = MyCustomGaussianModel()
+
+# Define energy and geometry for the dataset
+energy_axis = MapAxis.from_edges(np.logspace(-1.0, 2, 10), unit="TeV", name="energy", interp="log")
+geom = WcsGeom.create(skydir=(5.6, 0.2), width=5*u.deg, 
```

Comment:
maybe reduce a bit the overall size of the dataset to reduce its size? e.g. 4 degrees width should be enough